### PR TITLE
added custom input/output description support for e2

### PR DESCRIPTION
**Pull request by Divran**

### CODE REVIEW

**Comment by Divran** [comment]
image

**Comment by Divran** [comment]
```golo
@name memes
@inputs InputA #memes
@outputs OutputA #memes2
@inputs InputB #[ this is
a very long
multi-line
comment
yeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee ]#
@outputs OutputB #[this is a small inline comment]# OutputC

@inputs [InputC InputD #[woop]# InputE]:vector
```

**Comment by Divran** [comment]
Requesting reviewers here because of the way I implemented this could be questioned. I decided to go with "whenever an input or output definition is directly followed by a comment or block comment"

Other valid options include `@inputs Name:type (Description)` for example. Downside of other solutions would be that they require changes to syntax highlighting as well, whereas using comments doesn't.

But if you all think this is fine, then I can merge this tomorrow.

File: lua/entities/gmod_wire_expression2/base/preprocessor.lua [hunk 1]
```diff
@@ -43,12 +43,20 @@ function PreProcessor:HandlePPCommand(comment)
 end
 
 function PreProcessor:FindComments(line)
+	local isinput = not self.blockcomment and not self.multilinestring and line:match("^@inputs") ~= nil
```
Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/base/preprocessor.lua:46]: Does @inputs have to be the start of the line?

Comment (Divran) [reply lua/entities/gmod_wire_expression2/base/preprocessor.lua:46]: Yes

**Pull request merged**